### PR TITLE
[READY] Localnet configuration parity with staging

### PR DESCRIPTION
**Pull request by HebertCL**
# TL;DR

Configuration parity with staging and localnet.

## Summary

Added path-values file and descriptions to spin up and tear down localnet. Values file provides parity with the current staging deployment using same chart.

Additionally updated README to describe steps to bring up/down localnet and Makefile targets for it.

### CODE REVIEW

**Review by commoddity — COMMENTED** [review]
LGTM but just curious if you can give some more context on why we need to use Kind in addition to Tilt?

File: README.md
Comment (Olshansk) [inline README.md:302]: We have a pre-requisites section in `1.1`. Can you move it there?

File: Makefile
Comment (Olshansk) [inline Makefile:196]: I looked at the top of this Makefile and it seems like it's still using `docker compose`. 

Please merge & consolidate with `main`.

File: local/path/config/path-values.yaml
Comment (Olshansk) [inline local/path/config/path-values.yaml:1]: Why is this not part of the `helm-charts` repo?

File: README.md
Comment (Olshansk) [inline README.md:296]: See what we have in [1]

Once the user has gotten to step `8`, they don't want to relearn pre-requisites again, they just want to do that all upfront.

https://dev.poktroll.com/develop/developer_guide/quickstart.

File: Makefile
Comment (Olshansk) [inline Makefile:200]:
```suggestion

.PHONY: localnet_up
```

File: Makefile
Comment (Olshansk) [inline Makefile:201]: Any chance you can make this similar to what we have in `poktroll` for consistency?

```Makefile
.PHONY: localnet_up
localnet_up: check_docker_ps check_kind_context proto_regen localnet_regenesis ## Starts up a clean localnet
	tilt up
```

- Avoid inline flags
- Assume the cluster is already created
- Etc...

A couple other high level considerations:
- You can create a target called `make kind_config`  to do the things you're trying
- After you merge with main, check if there are any other better consolidations to be had.

File: README.md
Comment (Olshansk) [inline README.md:296]: Let's add the same explanation of why we need kind

cc @commoddity 

Screenshot 2024-12-03 at 7 45 43 PM

**Review by Olshansk — CHANGES_REQUESTED** [review]
@HebertCL GOing to take another look after the chances are addressed. 

1. Make sure to merge with main
2. High-level meta piece of feedback: try to approach the docs holistically so it seems like one whole wrote the whole thing.

It's not easy, but it goes a long way.

In this specific instance, the actionable is to consolidate pre-requisite secionts.

File: README.md
Comment (HebertCL) [reply README.md:301]: Moved the pre-requisites to deployment section.

File: Makefile
Comment (HebertCL) [reply Makefile:196]: I pulled main and verified the changes in main. The docker-compose section is still [there](https://github.com/buildwithgrove/path/blob/main/Makefile#L14-L28). I will remove those targets as well as the tilt ones in favor of the configuration I am pushing in this PR.

File: local/path/config/path-values.yaml
Comment (HebertCL) [reply local/path/config/path-values.yaml:1]: It IS part of the Path helm chart. 

The global directive encapsulates configurations that may spawn multiple times and across different resources. You can find references in [deployment](https://github.com/buildwithgrove/helm-charts/blob/main/charts/path/templates/deployment.yaml) and [serviceaccount](https://github.com/buildwithgrove/helm-charts/blob/main/charts/path/templates/serviceaccount.yaml) resources.

File: README.md
Comment (HebertCL) [reply README.md:295]: I've moved the pre-requisites to 1.1

File: Makefile
Comment (HebertCL) [reply Makefile:200]: This has been addressed.

**Comment by Olshansk** [comment]
@HebertCL Please re-request a review when this is ready.

File: Makefile
Comment (HebertCL) [reply Makefile]: I have split into `localnet.mk` to handle everything localnet-related.

File: README.md
Comment (HebertCL) [reply README.md:298]: Added a comment explaining the potential situation when using K8s Docker desktop.

**Review by Olshansk — APPROVED** [review]

**Comment by Olshansk** [comment]
@HebertCL I pushed a few changes but otherwise LGTM!

**Pull request merged**
